modules/bmc/ipmi.rb

```diff
@@ -5,22 +5,41 @@ module Proxy
   module BMC
     class IPMI < Base
       include Proxy::Log
+      attr_accessor :logger, :log_level
+
+      # set the log level of rubyipmi, set to 0 or Logger::DEBUG for usablable output
+      def self.log_level=(log_level)
+        Rubyipmi.log_level = log_level
+      end
+
+      def self.log_level
+        Rubyipmi.logger.level
+      end
+
+      def self.logger
+        Rubyipmi.logger
+      end
+
+      # set the logger for rubyipmi
+      def self.logger=(log)
+        Rubyipmi.logger = log
+      end
 
       def self.installed?(provider)
         # check with the lib to see if at least one provider is installed
-        if provider
+        begin
           Rubyipmi.is_provider_installed?(provider)
-        else
-          Rubyipmi.provider_installed?
+        rescue
+          false
         end
       end
 
-      def self.providers_installed?
-        return Rubyipmi.providers_installed?
+      def self.providers_installed
```

Comment:
This is a bit confusing -- the convention is that methods with a ```?``` suffix return a boolean, but here it appears to return a list...